rasa/shared/core/trackers.py

```diff
@@ -940,3 +940,33 @@ def get_trackers_for_conversation_sessions(
         )
         for evts in split_conversations
     ]
+
+
+def get_events_before_restart(
+    tracker: DialogueStateTracker,
+) -> DialogueStateTracker:
+    """Returns the tracker including all events before the first `restart` event.
+
+    If no `restart` is present, returns tracker with all events.
+
+    Args:
+        tracker: Instance of `DialogueStateTracker`.
+
+    Returns:
+        The tracker up until the first restart event.
+    """
+    split_conversations = events.split_events(
+        tracker.events,
+        Restarted,
+        include_splitting_event=False,
+    )
+    if len(split_conversations) == 0:
```

Comment:
We need some unit test coverage for this new function - I wasn't familiar with this `split_events` function, but it seems that it will always return a `List[List[Event]]` if there are events in the tracker, regardless whether there is a restart or not. The only time it will return an empty list is when the tracker has no events 🤔

Reply:
I'll take another look and add a test

Reply:
After looking at it again, I'm removing the fix. I think it requires more thought about what constitutes a "session" in the context of test stories. For example, should we be adding a record of the user's `/restart` utterance in test stories? What about `action_restart`? I think it's better to open a separate ticket and make decisions about these explicitly.